clientlibrary/worker/checkpointer.go

```diff
@@ -82,13 +83,12 @@ type DynamoCheckpoint struct {
 	Retries       int
 }
 
-func NewDynamoCheckpoint(dynamo dynamodbiface.DynamoDBAPI, kclConfig *config.KinesisClientLibConfiguration) Checkpointer {
+func NewDynamoCheckpoint(kclConfig *config.KinesisClientLibConfiguration) Checkpointer {
 	checkpointer := &DynamoCheckpoint{
```

Comment:
I'd rather not to change signature of the DynamoCheckpoiont constructor. Instead, splitting this file into two. checkpointer.go only contains interface and create another dynamodbcheckpointer.go for remaining.